lib/osproject_android.rb

```diff
@@ -45,6 +45,34 @@ def add_sdk!
       return
     end
 
+    application_class_created = false
+    application_class_exists = true
+    begin 
+      content = File.read(dir + '/' + app_class_location)
+    rescue
+      directory_split = app_class_location.split('/', -1)
+
+      application_name = directory_split[-1].split(".")[0]
+      com_index = directory_split.index "com"
+      range = directory_split.length - 2
+      package_directory = directory_split.slice(com_index..range)
```

Comment:
Looks like we are trying to get the package name we should use in Android application class? If so thinking we could simplify to `directory_split[0..-2]`. In ruby `-2` omits the last element in the array.

>  A negative index is assumed to be relative to the end of the array---that is, an index of -1 indicates the last element of the array, -2 is the next to last element in the array, and so on.

Reply:
Also code is also hard to understand what it is trying to do here. Could use helper methods or comments to understand this better.